deployment-audit/nmap/cybernet_target_audit.py

```
@@ -0,0 +1,378 @@
+        fields = sorted({k for row in rows for k in row}) if rows else []
+    with path.open("w", newline="", encoding="utf-8") as f:
+        writer = csv.DictWriter(f, fieldnames=list(fields), extrasaction="ignore")
+        writer.writeheader()
+        for row in rows:
+            writer.writerow(row)
+
+
+def parse_nmap_xml(path: Path) -> List[Dict[str, str]]:
+    if not path.exists():
+        return []
+    root = ET.parse(path).getroot()
```

Ruff (0.15.13)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)